draft-duke-quic-load-balancers.md

```diff
@@ -59,270 +61,545 @@ author:
-     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |  Type = 0x02  |
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |                    Unused Connection IDs                      |
-    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-                         Figure 2 ID_STOCK Message
-
-   This message simply reports the number of unused connection IDs in a
-   32-bit integer in Network order. Load Balancers MUST update their
-   estimate of server stock based on this message, as some connection
-   IDs may have been lost in transit.
-
-# Chained Load Balancers
-
-   In some deployments, there may be multiple tiers of trusted load
-   balancers in the path between client and server. All load balancers
-   using connection ID to encode routing information MUST agree on how
-   to decode connection IDs as routing instructions. Due to QUIC packet
-   authentication, connection IDs of established QUIC connections
-   cannot be rewritten in flight without access to the QUIC connection
-   keys.
-
-   A server configured to trust multiple load balancers MAY accept DTLS
-   connections from all of them and use provided Connection IDs
-   interchangeably. It SHOULD report its entire stock of connection IDs
-   to all trusted load balancers, rather than the number of IDs issued
-   from each source.
-
-# Security Considerations
-
+ 
+   The path between load balancer and server may pass through
+   middleboxes that could drop the coordination messages in this
+   protocol. It is therefore advantageous to make messages resemble
+   QUIC traffic as much as possible, as any viable path must obviously
+   admit QUIC traffic.
+
+## Load Balancer Chains
+
+   While it is possible to construct a scheme that supports multiple
+   low-state load balancers in the path, by using different parts of
+   the connection ID to encoding routing information for each load
+   balancer, this use case is out of scope for QUIC-LB.
+ 
+# Routing Algorithms
+ 
+   In QUIC-LB, load balancers do not send individual connection IDs to
+   servers. Instead, they communicate the parameters of one of two
+   algorithms to generate routable connection IDs.
+
+   The algorithms differ in the complexity of configuration at both
+   load balancer and server. Increasing complexity improves obfuscation
+   of the server mapping.
+
+   The load balancer SHOULD route Initial and 0-RTT packets from the
+   client using an alternate algorithm. Note that the SCID in these
+   packets may not be long enough to represent all the routing bits.
+   This algorithm SHOULD generate consistent results for Initial and
+   0RTT packets that arrive with the same source and destination
+   connection ID. The load balancer algorithms below apply to all
+   incoming Handshake and 1-RTT packets.
+
+## Plaintext CID Algorithm
```

Comment:
Before going into algorithm specifics at this point, I think the generic QUIC_LB packet format should be described and how it generally works. From there, explain the other type of QUIC_LB packets and then finally the algorithms that use these packets.

Reply:
I personally find it easier to follow to explain how to generate CIDs, since these are critical, and then talk about the many ways to distribute parameters, one of which includes QUIC-LB.

We can discuss later, but I'd like to move this commit forward.